lib/puppet_x/relay/util/http/client.rb

```diff
@@ -5,8 +5,12 @@ module Relay
     module Util
       module HTTP
         class Client
-          def initialize(base_url)
+          def initialize(base_url, settings = nil)
             @base_url = base_url
+            @proxy_host = settings[:proxy_host] if settings
+            @proxy_port = settings[:proxy_port] if settings
```

Comment:
It might be nice to raise an error here if `proxy_port` is defined but not `proxy_host`.

Reply:
Added, also added a spec test.